src/charm.py

```diff
@@ -912,21 +985,33 @@ def _unit_ip(self, unit: Unit) -> str:
 
     def get_secret(self, scope: str, key: str) -> Optional[str]:
         """Get secret from the secret storage."""
-        if scope == "unit":
+        if self._juju_has_secrets:
+            return self._juju_secret_get(scope, key)
+
+        if scope == UNIT_SCOPE:
             return self.unit_peer_data.get(key, None)
-        elif scope == "app":
+        elif scope == APP_SCOPE:
             return self.app_peer_data.get(key, None)
         else:
             raise RuntimeError("Unknown secret scope.")
 
-    def set_secret(self, scope: str, key: str, value: Optional[str]) -> None:
-        """Set secret in the secret storage."""
-        if scope == "unit":
+    def set_secret(self, scope: str, key: str, value: Optional[str]) -> Optional[str]:
+        """Set secret in the secret storage.
+
+        Juju versions > 3.0 use `juju secrets`, this function first checks
+          which secret store is being used before setting the secret.
+        """
+        if self._juju_has_secrets:
+            if not value:
+                return self._juju_secret_remove(scope, key)
+            return self._juju_secret_set(scope, key, value)
+
+        if scope == UNIT_SCOPE:
             if not value:
                 del self.unit_peer_data[key]
                 return
             self.unit_peer_data.update({key: str(value)})
-        elif scope == "app":
+        elif scope == APP_SCOPE:
             if not value:
                 del self.app_peer_data[key]
                 return
```

Comment:
This is just cosmetics, but why not:

```
        if not value:
            del self._peer_data[key]
            return
```

Reply:
(i.e. no `if` is needed)

Reply:
not sure what you mean by `if` is needed

Reply:
Also I prefer not change code related to old secrets at this PR